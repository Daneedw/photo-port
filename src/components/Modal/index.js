import React from 'react'

function Modal({description, category="landscape", name, closeModal, index=1}) {
    return (
        <div className="modalBackdrop">
        <div className="modalContainer">
          <h3 className="modalTitle">{name}</h3>
          <img alt="current category" src= {require(`../../assets/small/${category}/${index}.jpg`).default} />
          <p>
            {description}
          </p>
          <button type="button" onClick={closeModal}>
            Close this modal
          </button>
        </div>
      </div>
    )
}

export default Modal
