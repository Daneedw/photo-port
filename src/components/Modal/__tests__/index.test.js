import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Modal from '..'

const mockToggleModal = jest.fn();



const currentPhoto = {
  name: 'Park bench',
  category: 'landscape',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
  index: 1
};
afterEach(cleanup)

describe('Modal is rendering', () => {
  
  it('renders', () => {
    render(<Modal {...currentPhoto} closeModal={mockToggleModal}   />);
  });
  
  
  it('matches snapshot', () => {
    const { asFragment } = render(<Modal  />)
    expect(asFragment()).toMatchSnapshot()
  })
  
  describe('Click Event', () => {
    it('calls onClose handler', () => {
      // Arrange: Render Modal
      const { getByText } = render(<Modal
        {...currentPhoto}
        closeModal={mockToggleModal}
      />);
      // Act: Simulate click event
      fireEvent.click(getByText('Close this modal'));
      expect(mockToggleModal).toHaveBeenCalledTimes(1);
      // Assert: Expected matcher
    });
  })  

  // it('renders', () => {
  //   const { getByTestId } = render(<Modal currentCategory={portrait} />)
  //   expect(getByTestId('h1tag')).toHaveTextContent('Portraits')
  // })

})