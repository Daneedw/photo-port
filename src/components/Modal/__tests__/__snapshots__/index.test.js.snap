// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal is rendering matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      />
      <img
        alt="current category"
      />
      <p />
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
